[ERROR] : Error: Unknown error

Hi. Thanks for tool. But have some Unknown error when trying to get list of tasks from download station:

```
$ ./node_modules/.bin/syno --url http://USER:PASSWORD@192.168.0.37:5000 --debug dl getTaskList
[DEBUG] : Params URL detected : http://USER:PASSWORD@192.168.0.37:5000.
[DEBUG] : DSM Connection URL configured : http://USER:PASSWORD@192.168.0.37:5000
[DEBUG] : Account: USER
[DEBUG] : Password: PASSWORD
[DEBUG] : Host: 192.168.0.37
[DEBUG] : Port: 5000
[DEBUG] : API: 6.0
[DEBUG] : Ignore certificate errors: false
[DEBUG] : DSM Download Station API command selected
[DEBUG] : Method name configured : getTaskList
REQUEST { rejectUnauthorized: true,
  jar: true,
  json: true,
  url: 'http://192.168.0.37:5000/webapi/auth.cgi',
  qs: 
   { api: 'SYNO.API.Auth',
     version: 3,
     method: 'login',
     account: 'USER',
     passwd: 'PASSWORD',
     session: 'SYNO_SESSION_1478032259136' },
  callback: [Function] }
REQUEST make request http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST onRequestResponse http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:37 GMT',
  'content-type': 'text/plain; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  p3p: 'CP="IDC DSP COR ADM DEVi TAIi PSA PSD IVAi IVDi CONi HIS OUR IND CNT"',
  'set-cookie': [ 'id=wO12pG8X3GfQQ1470M4N227000;expires=Tue, 08-Nov-2016 20:20:37 GMT;path=/' ] }
REQUEST reading response's body
REQUEST finish init function http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST response end http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:37 GMT',
  'content-type': 'text/plain; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  p3p: 'CP="IDC DSP COR ADM DEVi TAIi PSA PSD IVAi IVDi CONi HIS OUR IND CNT"',
  'set-cookie': [ 'id=wO12pG8X3GfQQ1470M4N227000;expires=Tue, 08-Nov-2016 20:20:37 GMT;path=/' ] }
REQUEST end event http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST has body http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136 61
REQUEST emitting complete http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST { rejectUnauthorized: true,
  jar: true,
  json: true,
  url: 'http://192.168.0.37:5000/webapi/entry.cgi',
  qs: 
   { api: 'SYNO.DownloadStation2.Task.List',
     version: '1',
     method: 'get' },
  callback: [Function] }
REQUEST make request http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
REQUEST onRequestResponse http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:38 GMT',
  'content-type': 'application/json; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  'cache-control': 'max-age=0, no-cache, no-store, must-revalidate',
  pragma: 'no-cache',
  expires: '0' }
REQUEST reading response's body
REQUEST finish init function http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
REQUEST response end http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:38 GMT',
  'content-type': 'application/json; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  'cache-control': 'max-age=0, no-cache, no-store, must-revalidate',
  pragma: 'no-cache',
  expires: '0' }
REQUEST end event http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
REQUEST has body http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get 86
REQUEST emitting complete http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
[ERROR] : Error: Unknown error
REQUEST { rejectUnauthorized: true,
  jar: true,
  json: true,
  url: 'http://192.168.0.37:5000/webapi/auth.cgi',
  qs: 
   { api: 'SYNO.API.Auth',
     version: 3,
     method: 'logout',
     session: 'SYNO_SESSION_1478032259136' },
  callback: [Function] }
```

I have same error if try to make `listFiles`. The only method I get to work is `getInfo`.

Just have an idea: could it be related to node.js 7?

--- a/src/syno/Utils.coffee
+++ b/src/syno/Utils.coffee
@@ -76,8 +76,7 @@ class Utils
     # `str` [String]
     @listPluralize = (method, apiSubNname) ->
         if startsWith(method.toLowerCase(), 'list') and not endsWith(apiSubNname, 's')
-            lastWord = last(apiSubNname.split(/(?=[A-Z][^A-Z]+$)/))
-            apiSubNname = pluralize(lastWord) # pluralize if list
+            apiSubNname = apiSubNname.replace(/([A-Z][^A-Z]+)$/, (_, last) -> pluralize(last)) # pluralize if list
         return apiSubNname
         
     @createFunctionName = (apiName, method) ->

